datadog_checks_base/datadog_checks/checks/base.py

```diff
@@ -148,11 +149,13 @@ def _log_deprecation(self, deprecation_key):
     def service_check(self, name, status, tags=None, hostname=None, message=None):
         tags = self._normalize_tags_type(tags)
         if hostname is None:
-            hostname = ""
+            hostname = b''
         if message is None:
-            message = ""
+            message = b''
+        else:
+            message = ensure_bytes(message)
```

Comment:
Should we also encode the hostname? Not sure how common it is but I believe you can use unicode chars in it.

Reply:
I think hostnames are required to be ascii

https://tools.ietf.org/html/rfc952
https://tools.ietf.org/html/rfc1123#page-13

Reply:
Yup but the hostname can be overridden in different ways, for example in the config file, where a user could put a IDN.